pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -494,11 +526,9 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
             if (_vault.isPoolRegistered(childToken)) {
                 // Token is a BPT, so add liquidity to the child pool.
 
-                IERC20[] memory childPoolTokens = _vault.getPoolTokens(childToken);
-                (uint256[] memory childPoolAmountsIn, bool childPoolAmountsEmpty) = _getPoolAmountsIn(
-                    childPoolTokens,
-                    params.sender,
-                    params.wethIsEth
+                (uint256[] memory childPoolAmountsIn, bool childPoolAmountsEmpty) = _addLiquidityRecursive(
```

Comment:
I'm wondering if there's a drawback in this recursion. It's possible to refer to the same pool twice cyclicly, but that would work, probably (maybe a scenario that should be tested if we proceed with this).